controllers/database.go

```diff
@@ -20,26 +20,27 @@ import (
 	"sync"
```

Comment:
Maybe these should be moved to the `suite_test.go` considering these are only used for stubbing purposes.

Reply:
I wonder if they shouldn't go into internal/database ?

Reply:
Not sure, my first instinct was that they should be close to the tests where they are being used in. But you are right, this is an actual implementation that can even be an alternative to badger. With that in mind, I wonder if we should support toggling between the two implementations i.e. allow the user to configure using either the in memory ephemeral database storage or the persistent badger storage?

Reply:
Adding toggling feels like YAGNI just now, I can see a use-case for it, but it works with the simple image -> tags storage.

I'll move the in-memory version to the suite_test just now, but it's open to seeing it as a togglable replacement